incubator/hnc/third_party/open-policy-agent/gatekeeper/pkg/webhook/certs.go

```diff
@@ -54,23 +56,24 @@ func init() {
 	flag.BoolVar(&restartOnSecretRefresh, "cert-restart-on-secret-refresh", false, "Kills the process when secrets are refreshed so that the pod can be restarted (secrets take up to 60s to be updated by running pods)")
 }
 
-// AddRotator adds the CertRotator and ReconcileVWH to the manager.
-func AddRotator(mgr manager.Manager, cr *CertRotator, vwhName string) error {
+// AddRotator adds the CertRotator and ReconcileWH to the manager.
+func AddRotator(mgr manager.Manager, cr *CertRotator, vwhName string, crdNames []string) error {
 	cr.client = mgr.GetClient()
 	cr.certsNotMounted = make(chan struct{})
 	if err := mgr.Add(cr); err != nil {
 		return err
 	}
 
 	vwhKey := types.NamespacedName{Name: vwhName}
-	reconciler := &ReconcileVWH{
+	reconciler := &ReconcileWH{
 		client:    mgr.GetClient(),
 		scheme:    mgr.GetScheme(),
 		ctx:       context.Background(),
 		secretKey: cr.SecretKey,
 		vwhKey:    vwhKey,
+		crdNames:  crdNames,
 	}
-	if err := addController(mgr, reconciler, cr.SecretKey, vwhKey); err != nil {
+	if err := addController(mgr, reconciler, cr.SecretKey, vwhKey, crdNames); err != nil {
```

Comment:
Suggestion: change `addController` to accept the `ReconcileWH` type as its second parameter, instead of the `reconcile.Reconciler` interface. Using an interface doesn't buy us anything (the type is local; we don't mock it in the unit tests) and it would let us avoid passing the other three parameters since they're all in `ReconcileWH` anyway.

Reply:
Good sugguestion, updated!